src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -997,6 +1007,42 @@ void DecisionTree<FitnessFunction,
       probabilities);
 }
 
+//! Prune the tree to reduce complexity.
+template<typename FitnessFunction,
+         template<typename> class NumericSplitType,
+         template<typename> class CategoricalSplitType,
+         typename DimensionSelectionType,
+         bool NoRecursion>
+bool DecisionTree<FitnessFunction,
+                  NumericSplitType,
+                  CategoricalSplitType,
+                  DimensionSelectionType,
+                  NoRecursion>::Prune(double threshold)
+{
+
+  size_t numChildren = node->NumChildren();
+
+  for(size_t i = 0; i < numChildren; ++i)
+  {
+    bool store = numChildren[i]->Prune(threshold);
+    if(store == true)
+    {
+      children.erase(children.begin() + i);
```

Comment:
Don't you have to decrease i in this case as well? i.e.

a = [1, 2, 3, 4]

```
i = 0
erase(begin() + 0)
i = 1
erase(begin() + 1) <-- should be 0, since we removed the first element already?
i = 2
i = 3
```

Reply:
Ah yes... I wrote that initially then deleted it for some reason 🤦🏻